pkg/controller/machinedeployment/sync.go

```diff
@@ -111,7 +111,7 @@ func (dc *MachineDeploymentControllerImpl) getNewMachineSet(d *v1alpha1.MachineD
 
```

Comment:
`SetDeploymentRevision` one line above this comment where I can unfortunatelly not comment updates an annotation if required: https://github.com/kubernetes-sigs/cluster-api/blob/3d9767cdbbbb9f5c7473e6130df5d9e1730a43e0/pkg/controller/machinedeployment/util/util.go#L115

Consequently we must update the `MachienDeployment`, not its status.

The current behaviour results in the `UpdateStatus` being basically a no-op, except it triggers the informer, which then triggers the `sync`, which triggers this `UpdateStatus` again etc.